File: backend/backend/domain/repositories/process_repository.py
"""ProcessRepository 인터페이스."""

from __future__ import annotations

from abc import abstractmethod

from backend.domain.entities.process import Process, WorkType
from backend.domain.repositories.base import BaseRepository


class ProcessRepository(BaseRepository[Process, str]):
    """공정 Repository 인터페이스."""

    @abstractmethod
    async def get_by_work_type(self, work_type: WorkType) -> list[Process]:
        """작업 유형별 공정 조회."""
        ...

File: frontend/src/domain/repositories/process.repository.ts
/**
 * Process Repository Interface
 * 공정 데이터 접근 인터페이스
 */

import type { Process, ProcessCreateInput, ProcessUpdateInput, WorkType } from '../entities';
import type { BaseRepository } from './base.repository';

export interface IProcessRepository
  extends BaseRepository<Process, ProcessCreateInput, ProcessUpdateInput> {
  /**
   * 작업 유형별 조회 (내작/외작)
   */
  getByWorkType(workType: WorkType): Promise<Process[]>;

  /**
   * 공정 일괄 수정
   */
  bulkUpdate(processes: Array<{ id: string } & ProcessUpdateInput>): Promise<Process[]>;

  /**
   * 공정 검색 (공정코드, 공정명)
   */
  search(query: string): Promise<Process[]>;
}
